src/omv/ports/stm32/jpeg.c

```diff
@@ -0,0 +1,753 @@
+
+        int dy = src->h - y_offset;
+        if (dy > MCU_H) {
+            dy = MCU_H;
+        }
+
+        for (int x_offset = 0; x_offset < src->w; x_offset += MCU_W) {
+            int8_t *Y0 = (int8_t *) (mcu_row_buffer_ptr + (mcu_size * (x_offset / MCU_W)));
+            int8_t *CB = Y0 + JPEG_444_GS_MCU_SIZE;
+            int8_t *CR = CB + JPEG_444_GS_MCU_SIZE;
+
+            int dx = src->w - x_offset;
+            if (dx > MCU_W) {
+                dx = MCU_W;
+            }
+
+            // Copy 8x8 MCUs.
+            jpeg_get_mcu(src, x_offset, y_offset, dx, dy, Y0, CB, CR);
+        }
+
+        // Flush the MCU row for DMA...
+        SCB_CleanDCache_by_Addr((uint32_t *) mcu_row_buffer_ptr, src_w_mcus_bytes);
+
+        if (!y_offset) {
+            // Invalidate the output buffer.
+            SCB_InvalidateDCache_by_Addr(dma_buffer, OUTPUT_CHUNK_SIZE);
+            // Start the DMA process off on the first row of MCUs.
+            HAL_JPEG_Encode_DMA(&JPEG_Handle, mcu_row_buffer_ptr, src_w_mcus_bytes, dma_buffer, OUTPUT_CHUNK_SIZE);
+        } else {
+
+            // Wait for the last row MCUs to be processed before starting the next row.
+            while (!JPEG_input_paused) {
+                __WFI();
+
+                if (JPEG_output_paused) {
+                    memset(&JPEG_Config, 0, sizeof(JPEG_ConfTypeDef));
+                    HAL_JPEG_Abort(&JPEG_Handle);
+                    fb_free(); // mcu_row_buffer (after DMA is aborted)
+                    return true; // overflow
+                }
+            }
+
+            // Reset the lock.
+            JPEG_input_paused = false;
+
+            // Restart the DMA process on the next row of MCUs (that were already prepared).
+            HAL_JPEG_ConfigInputBuffer(&JPEG_Handle, mcu_row_buffer_ptr, src_w_mcus_bytes);
+            HAL_JPEG_Resume(&JPEG_Handle, JPEG_PAUSE_RESUME_INPUT);
+        }
+    }
+
+    // After writing the last MCU to the JPEG core it will eventually generate an end-of-conversion
+    // interrupt which will finish the JPEG encoding process and clear the busy flag.
+
+    while (HAL_JPEG_GetState(&JPEG_Handle) == HAL_JPEG_STATE_BUSY_ENCODING) {
+        __WFI();
+
+        if (JPEG_output_paused) {
+            memset(&JPEG_Config, 0, sizeof(JPEG_ConfTypeDef));
+            HAL_JPEG_Abort(&JPEG_Handle);
+            fb_free(); // mcu_row_buffer (after DMA is aborted)
+            return true; // overflow
+        }
+    }
+
+    fb_free(); // mcu_row_buffer
+
+    // Set output size.
+    dst->size = JPEG_out_data_length;
+
+    // STM32H7 BUG FIX! The JPEG Encoder will occasionally trigger the EOCF interrupt before writing
+    // a final 0x000000D9 long into the output fifo as the end of the JPEG image. When this occurs
+    // the output fifo will have a single 0 value in it after the encoding process finishes.
+    if (__HAL_JPEG_GET_FLAG(&JPEG_Handle, JPEG_FLAG_OFNEF) && (!JPEG_Handle.Instance->DOR)) {
```

Comment:
Are you sure about this ? Note you were also very sure about the other one.

Reply:
Yes, this definitely occurs.